### ISSUE TITLE
Cargo index update stopped working with url rewrite

### ISSUE THREAD

**Issue by mriehl** [nabijaczleweli/cargo-update#100]
Hi,

`cargo install-update` stopped working for me:

```bash
cargo install-update -a
    Updating registry 'https://github.com/rust-lang/crates.io-index'
Failed to update index repository: authentication required but no callback set.
```

Cargo itself is still able to update the crates index.

I managed to track down the culprit, in my `.gitconfig` I have:
```
[url "git@github.com:"]                                                                                                                                                                                                                                                                                                                             
 insteadOf = https://github.com/  
```
I think I added this because of golang's `go get` that hardcodes https urls, while I use almost only ssh agent git remoting.
Anyway this rewrites the https crates repository to ssh, which then requires a callback challenge (ssh agent), which isn't set by cargo-update.


Cargo itself is able to deal with this because they add the git2 callbacks for remote authentication (ssh agent in this case).
Would you consider adding callbacks too, so that it "just works" over ssh too? I can try to hack something together.

For reference [this is cargo's credential callback](https://github.com/rust-lang/cargo/blob/bb28e71202260180ecff658cd0fa0c7ba86d0296/src/cargo/sources/git/utils.rs#L344-L391)

**Comment by nabijaczleweli** [comment]
Hey, sorry for the delay on this, but I was in absolutely no state to even read this, not that I fully get it now, when words actually seem to have meaning :v

So, like, that piece of `.gitconfig` silently rewrites all URLs to use SSH, and using SSH requires some additional "challenge" callback which we don't set?

If you could slap something together, that'd be great, yes 💜

**Comment by nabijaczleweli** [comment]
@mriehl can you check current `master` with the commit above?

**Comment by nabijaczleweli** [comment]
Released in [`v1.8.0`](https://github.com/nabijaczleweli/cargo-update/releases/tag/v1.8.0), but I'm all ears for how well this behaves!
